includes/process-purchase.php

```diff
@@ -1064,3 +1066,28 @@ function edd_process_straight_to_gateway( $data ) {
 	edd_send_to_gateway( $purchase_data['gateway'], $purchase_data );
 }
 add_action( 'edd_straight_to_gateway', 'edd_process_straight_to_gateway' );
+
+/**
+ * Update the user first and last name
+ *
+ * @since
+ * @param array $user_data
+ *
+ * @return int|WP_Error
+ */
+function edd_update_user( $user_data = array() ) {
+	// Bail, if we don't have a user to update
+	if ( ! isset( $user_data['user_id'] ) ) {
+		return false;
+	}
+
+	$user_args = array(
```

Comment:
This would be better as:
$user_args = array( );

if ( isset( $user_data['user_first'] )  ){
     $user_args['first_name'] = $user_data['user_first']; 
}
// same thing for last name here

if ( !empty( $user_args ) ){
  // add id to array
  // update user here
} 

That way the user only gets updated if and only if they have new information coming in (we don't want to lose old data), and if we only have a first name only update that, not last_name if we don't have that.